addons/sale_timesheet/data/sale_service_demo.xml

```diff
@@ -158,18 +177,145 @@
+            <value eval="{'user_ids': [(4, ref('base.user_demo'))]}"/>
+        </function>
+
+        <!-- Tasks progress -->
+        <function model="project.task" name="write">
+            <value model="project.task" search="[('sale_line_id', '=', ref('sale_line_11'))]"/>
+            <value eval="{'stage_id': ref('project.project_stage_1')}"/>
+        </function>
+        <function model="project.task" name="write">
+            <value model="project.task" search="[('sale_line_id', '=', ref('sale_line_13'))]"/>
+            <value eval="{'stage_id': ref('project.project_stage_1')}"/>
+        </function>
+        <function model="project.task" name="write">
+            <value model="project.task" search="[('sale_line_id', '=', ref('sale_line_21'))]"/>
+            <value eval="{'stage_id': ref('project.project_stage_1')}"/>
+        </function>
+        <function model="project.task" name="write">
+            <value model="project.task" search="[('sale_line_id', '=', ref('sale_line_22'))]"/>
+            <value eval="{'stage_id': ref('project.project_stage_1')}"/>
+        </function>
+
+        <!-- Personal stages for SOL tasks -->
+        <function model="project.task.stage.personal" name="write">
+            <value model="project.task.stage.personal" search="[('task_id', '=', obj().env['project.task'].search([('sale_line_id', '=', ref('sale_line_13'))]).id), ('user_id', '=', ref('base.user_admin'))]"/>
+            <value eval="{'stage_id': ref('project.project_personal_stage_admin_2')}"/>
+        </function>
+        <function model="project.task.stage.personal" name="write">
+            <value model="project.task.stage.personal" search="[('task_id', '=', obj().env['project.task'].search([('sale_line_id', '=', ref('sale_line_22'))]).id), ('user_id', '=', ref('base.user_demo'))]"/>
+            <value eval="{'stage_id': ref('project.project_personal_stage_demo_1')}"/>
+        </function>
+
+        <!-- Activities for SOL tasks -->
+        <record id="project_task_agr_activity_1" model="mail.activity">
+            <field name="res_id" model="project.task" search="[('sale_line_id', '=', ref('sale_line_13'))]"/>
+            <field name="res_model_id" ref="project.model_project_task"/>
+            <field name="activity_type_id" ref="mail.mail_activity_data_upload_document"/>
+            <field name="date_deadline" eval="(DateTime.today() + relativedelta(days=2)).strftime('%Y-%m-%d %H:%M')"/>
+            <field name="summary">Upload new plans</field>
+            <field name="create_uid" ref="base.user_admin"/>
+            <field name="user_id" ref="base.user_admin"/>
+        </record>
+
+        <!-- Assign Support tasks to Marc Demo -->
+        <function model="project.task" name="write">
+            <value model="project.task" search="[('sale_line_id', '=', ref('sale_line_11'))]"/>
+            <value eval="{'user_ids': [(4, ref('base.user_demo'))]}" />
```

Comment:
I don't know if it works but I think we could use the Command class instead of magic number. 🙂 
```suggestion
            <value eval="{'user_ids': [Command.link(ref('base.user_demo'))]}" />
```

Reply:
It worked, done :+1: